diesel/src/migrations/migration.rs

```diff
@@ -97,6 +98,31 @@ pub fn version_from_path(path: &Path) -> Result<String, MigrationError> {
         })
 }
 
+#[allow(missing_debug_implementations)]
+pub struct MigrationName<'a> {
+    pub migration: &'a Migration,
+}
+
+pub fn name(migration: &Migration) -> MigrationName {
+    MigrationName {
+        migration: migration,
+    }
+}
+
+impl<'a> fmt::Display for MigrationName<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Some(file_path) = self.migration.file_path() {
+            if let Some(file) = file_path.file_name() {
+                if let Some(name) = file.to_str() {
```

Comment:
How about this:

```rust
let file_name = self.migration.file_path()
    .and_then(|file| file_path.file_name())
    .and_then(|file| file.to_str());
if let Some(name) = file_name {
    f.write_str(name)?;
}
Ok(())
```

Reply:
Also this needs to fall back to printing the version if no file path is present. Embedded migrations return `None` for the file path

Reply:
@juliusdelta What do you think about changing this pyramid to use `and_then` like my comment above?

Reply:
@sgrif I had but changed it back for some reason, and I can't quite remember why. Anyway, since an embedded migration would return `None` for `file_path()` I can just add an `else` like:

```rust
impl <'a> fmt::Display for MigrationName<'a> {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let file_name = self.migration.file_path()
            .and_then(|file| file_path.file_name())
            .and_then(|file| file.to_str());
        if let Some(name) = file_name {
            f.write_str(name)?;
        } else {
            f.write_str(self.migration.version())?;
        }
        Ok(())
    }
}
```
to handle that case. Right?

Reply:
Yes

Reply:
Or if you wanted to get really fancy:

```rust
impl <'a> fmt::Display for MigrationName<'a> {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let file_name = self.migration.file_path()
            .and_then(|file| file_path.file_name())
            .and_then(|file| file.to_str())
            .map(|name| f.write_str(name))
            .unwrap_or_else(|| f.write_str(self.migraiotn.version))
    }
}

Reply:
Fancy, you say? Then by all means, try and go with this!

```rust
let file_name = self.migration.file_path()
    .and_then(Path::file_name)
    .and_then(OsStr::to_str) // ...
```